File: verification/hash.py
#!/usr/bin/env python3
"""Thin SHA-256 helper. Leaf construction = SHA-256(canonical_bytes)."""
from __future__ import annotations
import hashlib
from typing import Union

def sha256(data: Union[bytes, bytearray, memoryview]) -> bytes:
    """Return 32-byte digest."""
    return hashlib.sha256(data).digest()

def sha256_hex(data: Union[bytes, bytearray, memoryview]) -> str:
    return hashlib.sha256(data).hexdigest()

if __name__ == "__main__":
    import sys
    raw = sys.stdin.buffer.read()
    print(sha256_hex(raw))

File: verification/hash.mjs
#!/usr/bin/env node
/** Thin SHA-256 helper. Leaf construction = SHA-256(canonical_bytes). */
import { createHash } from "node:crypto";
import { readFileSync } from "node:fs";

export function sha256(data) {
  return createHash("sha256").update(data).digest();
}

export function sha256Hex(data) {
  return createHash("sha256").update(data).digest("hex");
}

if (import.meta.url === `file://${process.argv[1]}`) {
  const buf = process.argv[2]
    ? readFileSync(process.argv[2])
    : readFileSync(0);
  process.stdout.write(sha256Hex(buf) + "\n");
}
